### PR TITLE
Updated endpoints

### PR DESCRIPTION
**Pull request by matthewleemle**
Changed routing to match specifications, added JWT configurations, abstracted secret keys into env files, user info route updated to check if id matches with JWT token, login route now returns automatically generated ID of user

### CODE REVIEW

File: Controllers/users.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
+require('dotenv').config({path: __dirname + '/process.env'})
```
Comment (Asiapenguin) [inline Controllers/users.js]: is there a reason for you to use the filename process.env instead of .env?

File: Controllers/users.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      success: true,
+      message: 'Authentication successful!',
+      token: token,
+      id: result._id
+      })
+    console.log("Successful login.");
+    }
+
+  else{
+    res.status(404).send("Failed login");
```
Comment (Asiapenguin) [inline Controllers/users.js]: In the case with wrong password/credentials, the status code should be 401 Unauthorized

File: Controllers/users.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      "username":username,
+      "email":email,
+      "password":pass,
+      "phone":phone
+      }
+
+    db.collection('details').insertOne(data,function(err, collection){
+      if (err) throw err;
+      console.log("Record inserted Successfully");
+      res.status(200).send("Successful signup");
```
Comment (Asiapenguin) [inline Controllers/users.js]: this should return a json containing user information, of course without password

File: Controllers/users.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    });
+}
+
+async function userInfo(req,res){
+  const result = await db.collection('details').findOne({username: req.decoded.username});
+  if(req.params.id==result._id){
+      res.status(200).send(result);
+  }
+  else{
+    res.status(400).send("Access not allowed.");
```
Comment (Asiapenguin) [inline Controllers/users.js]: this should be 404 Not Found. The client who made the request gave an id that does not exist on our database, thus not found.

File: Controllers/users.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+let jwt = require("jsonwebtoken");
+let middleware = require("../middleware");
+
+async function loginUser(req,res){
+  var username = req.body.username;
+  var password = req.body.password;
+
+  var data = {
+      "username": username,
+      "password":password,
```
[the remark this answers is not in the document]
Comment (matthewleemle) [reply Controllers/users.js]: what will we use to hash the password? Is there a specific API to use or?

File: Controllers/users.js (hunk 4 above)
Comment (matthewleemle) [reply Controllers/users.js]: What I thought was if in this case the user gave an ID thats valid but not supplying the correct login jwt token that contains the username that corresponds to the ID, but now that i think about it its unlikely that happens

File: Controllers/users.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (Asiapenguin) [reply Controllers/users.js]: Look into "bcrypt"

File: Controllers/users.js (hunk 4 above)
Comment (Asiapenguin) [reply Controllers/users.js]: yea its unlikely

**Pull request merged**